server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -837,6 +861,45 @@ private IndexMetadata updateIndexSettings(
                         return builder.settings(settingsBuilder).build();
                     }
 
+                    private void validateSearchableSnapshotRestorable(long totalRestorableRemoteIndexesSize) {
+                        ClusterInfo clusterInfo = clusterInfoSupplier.get();
+                        double remoteDataToFileCacheRatio = DATA_TO_FILE_CACHE_SIZE_RATIO_SETTING.get(clusterService.getSettings());
+                        Map<String, FileCacheStats> nodeFileCacheStats = clusterInfo.getNodeFileCacheStats();
+                        if (nodeFileCacheStats.isEmpty() || remoteDataToFileCacheRatio <= 0.0f) {
+                            return;
+                        }
+
+                        long totalNodeFileCacheSize = clusterInfo.getNodeFileCacheStats()
+                            .values()
+                            .stream()
+                            .map(fileCacheStats -> fileCacheStats.getTotal().getBytes())
+                            .mapToLong(Long::longValue)
+                            .sum();
+
+                        Predicate<ShardRouting> shardRoutingPredicate = shardRouting -> shardRouting.primary()
+                            && indicesService.indexService(shardRouting.index()).getIndexSettings().isRemoteSnapshot();
+
+                        ShardsIterator shardsIterator = clusterService.state()
+                            .routingTable()
+                            .allShardsSatisfyingPredicate(shardRoutingPredicate);
```

Comment:
Nitpick, maybe the following reads better?

```suggestion
                        Predicate<ShardRouting> isRemoteSnapshotShard = shardRouting -> shardRouting.primary()
                            && indicesService.indexService(shardRouting.index()).getIndexSettings().isRemoteSnapshot();

                        ShardsIterator shardsIterator = clusterService.state()
                            .routingTable()
                            .allShardsSatisfyingPredicate(isRemoteSnapshotShard);
```

Reply:
Updated.